### ISSUE TITLE
Failed to save "test.erl". The content on disk is newer.

### ISSUE THREAD

**Issue by areel** [pgourlain/vscode_erlang#94]
Hi

When I have this extension loaded after a small number of successful saves (2-3) I receive the following error message:
Failed to save 'filename.erl': The content on disk is newer. Please compare your version with the one on disk.

Once this error appears it will reappear for subsequent saves.

This occurs on my windows 10 and ubuntu dev environments. 
VS: 1.24.0
erlang 0.4.4

Cheers

Aidan

**Comment by pgourlain** [comment]
Hi,

Thanks for your report !

Do you have same issue with 'AutoSave' to true  ?

**Comment by areel** [comment]
Thinking it was race condition related to the language server, I have been testing by varying the processor count on the ubuntu vm,
I have reduced the processor count down from 4 to 1 and issue did not repeat.
I then upped the count to 16 and issue did not repeat itself.
I then went back to the original and issue did not repeat itself

I will leave the autosave option off (as it was), when/if the issue resurfaces I will enable tracing and forward to you. Where are the trace files stored?

**Comment by pgourlain** [comment]
Traces are displayed in 'Output' Pane, and choose 'Erlang Language Server' view

**Comment by wojteksurowka** [comment]
Is the issue still reproducible?

**Comment by areel** [comment]
Yes, I had it occur again, but didnt capture the error logging , I tried autosave which resolved it and I continued with my work. I will switch off autosave and forward the logging later today or tomorrow.

**Comment by areel** [comment]
Hi, attached is a file containing a session where I started the editor, done one change with successful save, then a second edit and received the error message.

[4-startup-FirstSaveSuccessful-SecondSaveFailing.txt](https://github.com/pgourlain/vscode_erlang/files/2112384/4-startup-FirstSaveSuccessful-SecondSaveFailing.txt)

**Comment by wojteksurowka** [comment]
@areel in the logs i see formatting request as the last one. Do you see any correlation between using Format Document and the error you observe? Right now Format is the only operation performed by the language server which modifies the original file.

**Comment by wojteksurowka** [comment]
Fix for Format Document is in pull request #99 - the file on disk should no longer be modified by formatting directly outside of control of VSC. Version 0.4.6 should include the fix.
